src/webview/java/org/medicmobile/webapp/mobile/EmbeddedBrowserActivity.java

```diff
@@ -40,15 +40,21 @@
 
 @SuppressWarnings({ "PMD.GodClass", "PMD.TooManyMethods" })
 public class EmbeddedBrowserActivity extends LockableActivity {
-	/** Any activity result with all 3 low bits set is _not_ a simprints result. */
+	/**
+	 * Any activity result with all 3 low bits set is _not_ a simprints result.
+	 *
+	 * The following block of bit-shifted integers are intended for use in the subsystem seen
+	 * in the onActivityResult below. These integers respect the reserved block of integers
+	 * which are used by simprints. Simprint intents are started in the webapp where a matching
+	 * bitmask is used to respect the scheme on that side of things.
+	 * */
 	private static final int NON_SIMPRINTS_FLAGS = 0x7;
 	static final int GRAB_PHOTO = (0 << 3) | NON_SIMPRINTS_FLAGS;
 	static final int GRAB_MRDT_PHOTO = (1 << 3) | NON_SIMPRINTS_FLAGS;
-	private static final int ACCESS_FINE_LOCATION_PERMISSION_REQUEST = (int)(Math.random() * 1000);
+	static final int DISCLOSURE_LOCATION_PERMISSION_REQUEST = (2 << 3) | NON_SIMPRINTS_FLAGS;
 
-	// Use different ranges of random values for other request codes caught by the same method,
-	// eg. (int)(Math.random() * 1000) + 1000
-	private static final int DISCLOSURE_LOCATION_PERMISSION_REQUEST = (int)(Math.random() * 1000);
+	// Arbitrarily selected value
+	private static final int ACCESS_FINE_LOCATION_PERMISSION_REQUEST = 7038678;
```

Comment:
I think some of the confusion here is that these codes look kind of similar but they are used in different ways. The first three are used for activities and the last one is used for a permission request. In particular `DISCLOSURE_LOCATION_PERMISSION_REQUEST` should be renamed because it's _not_ a permission request, it's launching an activity. This will help give a clue as to why the first three need to be masked to not conflict with simprints but the last one doesn't, and hopefully help this not happen again!

I think they should be renamed as follows:

`GRAB_PHOTO` -> `GRAB_PHOTO_ACTIVITY_REQUEST_CODE`
`GRAB_MRDT_PHOTO` -> `GRAB_MRDT_PHOTO_ACTIVITY_REQUEST_CODE`
`DISCLOSURE_LOCATION_PERMISSION_REQUEST` -> `DISCLOSURE_LOCATION_ACTIVITY_REQUEST_CODE`
`ACCESS_FINE_LOCATION_PERMISSION_REQUEST` -> `ACCESS_FINE_LOCATION_PERMISSION_REQUEST_CODE`